xwiki-platform-core/xwiki-platform-extension/xwiki-platform-extension-handlers/xwiki-platform-extension-handler-xar/src/main/java/org/xwiki/extension/xar/internal/security/XarSecurityRule.java

```diff
@@ -78,4 +80,28 @@ public RuleState getState()
     {
         return RuleState.DENY;
     }
+
+    @Override
+    public int hashCode()
+    {
+        return new HashCodeBuilder().append(right).append(simple).append(securityTool).toHashCode();
+    }
+
+    @Override
+    public boolean equals(Object object)
+    {
+        if (object == this) {
+            return true;
+        }
+        if (!(object instanceof XarSecurityRule)) {
+            return false;
+        }
+
+        XarSecurityRule rhs = (XarSecurityRule) object;
+        return new EqualsBuilder()
+                .append(this.right, rhs.right)
+                .append(this.simple, rhs.simple)
+                .append(this.securityTool, rhs.securityTool)
```

Comment:
This is not really relevant IMO. It's not data of that rules, just an internal helper.

Reply:
Fixed in  d712af2